numpy/core/tests/test_umath_accuracy.py

```diff
@@ -50,5 +50,7 @@ def test_validate_transcendentals(self):
                         perm = np.random.permutation(len(inval))
                         inval = inval[perm]
                         outval = outval[perm]
+                        inval[np.argwhere(np.isnan(inval))] = np.nan
+                        outval[np.argwhere(np.isnan(inval))] = np.nan
```

Comment:
a comment here would be helpful

Reply:
Can we fix this within the `assert_array_max_ulp` function? That function should be consistent in either rejecting or accepting NaN I think.

Reply:
`assert_array_max_ulp`  indeed does not document its NaN behaviour, so I guess we can document and fix it there.

Reply:
also it is a bit annoying that `assert_array_max_ulp` does not show the offending output like `assert_array_compare` but only the maximum ULP discovered.

Reply:
makes sense, I can update the `assert_array_max_ulp ` function instead. Where would you like me to add the document?

Reply:
the docstring of the function itself does not indicate any special handling of NaNs. If we change that, we should document it

Reply:
in the docstring